src/lib/authui/ldapkrb/main_dialog.rb

```diff
@@ -306,6 +306,12 @@ def save_ldap
             when :ldap_tls_method_starttls
                 AuthConfInst.ldap_conf['ssl'] = 'start_tls'
             end
+
+            # bsc#1162025: Default bind_policy to soft if not present.
+            if not AuthConfInst.ldap_conf.key?('bind_policy')
```

Comment:
Although here it will work as you expect, in general I'd avoid using the `not`, `and`, and `or` operators in Ruby, which has a quite low precedence. [This article](https://mixandgo.com/learn/understanding-boolean-operator-precedence-in-ruby) provides some examples about it.

The line could be rewritten as 


```ruby
AuthConfInst.ldap_conf['bind_policy'] = 'soft' unless AuthConfInst.ldap_conf.key?('bind_policy')
```